File: jmolecules-axon/02-axonstyle-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/query/MoneyTransfer.java
package org.jmolecules.example.axonframework.domain.api.query;

public record MoneyTransfer(
  String moneyTransferId,
  String sourceAccountId,
  String targetAccountId,
  int amount,
  Boolean success,
  String message
) {

  public MoneyTransfer(String moneyTransferId, String sourceAccountId, String targetAccountId, int amount) {
    this(moneyTransferId, sourceAccountId, targetAccountId, amount, null, null);
  }

  public MoneyTransfer(String moneyTransferId, String sourceAccountId, String targetAccountId, int amount, String message) {
    this(moneyTransferId, sourceAccountId, targetAccountId, amount, message == null, message);
  }

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/model/moneytransfer/MoneyTransfer.kt
package org.jmolecules.example.axonframework.domain.model.moneytransfer

import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.Amount
import org.jmolecules.example.axonframework.domain.api.type.MoneyTransferId

data class MoneyTransfer(
    val moneyTransferId: MoneyTransferId,
    val sourceAccountId: AccountId,
    val targetAccountId: AccountId,
    val amount: Amount
)
